composer/models/loss.py

```diff
@@ -199,20 +228,31 @@ def soft_cross_entropy(input: Tensor,
         assert reduction in ['sum', 'mean', 'none'], f"{reduction} reduction not supported."
         assert size_average is None, "size_average is deprecated"
         assert reduce is None, "reduce is deprecated"
-        assert ignore_index == -100, "ignore_index not supported."
-        probs = -1 * (target * F.log_softmax(input, dim=1))
+        if ignore_index != -100:
+            warnings.warn("ignore_index not supported when using dense labels. Ignoring targets with 0 probability.")
+        xentropy = -1 * (target * F.log_softmax(input, dim=1))
```

Comment:
skip the broadcast-multiply?
```suggestion
        xentropy = -(target * F.log_softmax(input, dim=1))
```

Reply:
Done!